File: api/src/main/java/br/com/vr/apis/requests/CreatePurchaseCardRequest.java
package br.com.vr.apis.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class CreatePurchaseCardRequest {

    @JsonProperty("cnpj") private String cnpj;
    @JsonProperty("purchaseCardType") private String purchaseCardType;
    @JsonProperty("category") private String category;
    @JsonProperty("cashBack") private Integer cashBack;
    @JsonProperty("pat") private Boolean pat;

}

File: api/src/main/kotlin/br/com/vr/apis/requests/CreatePurchaseCardRequest.kt
package br.com.vr.apis.requests

import com.fasterxml.jackson.annotation.JsonProperty

data class CreatePurchaseCardRequest(
    @JsonProperty("cnpj")
    val cnpj: String?,
    @JsonProperty("purchaseCardType")
    val purchaseCardType: String?,
    @JsonProperty("category")
    val category: String? ,
    @JsonProperty("cashBack")
    val cashBack: Int?,
    @JsonProperty("pat")
    val pat: Boolean?
)
